mne/viz/_figure.py

```diff
@@ -105,8 +105,10 @@ def _close(self, event):
     def _keypress(self, event):
         """Handle keypress events."""
         if event.key == self.mne.close_key:
-            from matplotlib.pyplot import close
+            from matplotlib.pyplot import close, rcParams
             close(self)
+            if rcParams['backend'].lower() == 'agg':
+                self._close(event)
```

Comment:
I'm not crazy about having backend-triaging code paths in the main codebase (less objection if it's in the tests)

Reply:
Hmmm, okay I can just call _close from the test, we can try that